pkg/controllers/internalmembercluster/v1beta1/member_controller_test.go

```diff
@@ -1178,3 +1185,296 @@ func TestUpdateResourceStats(t *testing.T) {
+		propertyProvider propertyprovider.PropertyProvider
+		wantIMCStatus    clusterv1beta1.InternalMemberClusterStatus
+	}{
+		{
+			name:             "provider failed to start",
+			propertyProvider: &failedToStartProvider{},
+			wantIMCStatus: clusterv1beta1.InternalMemberClusterStatus{
+				Properties: map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+					propertyprovider.NodeCountProperty: {
+						Value: "1",
+					},
+				},
+				ResourceUsage: clusterv1beta1.ResourceUsage{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("10"),
+						corev1.ResourceMemory: resource.MustParse("10Gi"),
+					},
+					Allocatable: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("8"),
+						corev1.ResourceMemory: resource.MustParse("8Gi"),
+					},
+					Available: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("5"),
+						corev1.ResourceMemory: resource.MustParse("5Gi"),
+					},
+				},
+				Conditions: []metav1.Condition{
+					{
+						Type:               string(clusterv1beta1.ConditionTypeClusterPropertyProviderStarted),
+						Status:             metav1.ConditionFalse,
+						Reason:             ClusterPropertyProviderStartedFailedReason,
+						Message:            fmt.Sprintf(ClusterPropertyProviderStartedFailedMessage, "expected to fail"),
+						ObservedGeneration: imcTemplate.GetGeneration(),
+					},
+				},
+			},
+		},
+		{
+			name: "no provider is set up, fall back to the default behavior",
+			wantIMCStatus: clusterv1beta1.InternalMemberClusterStatus{
+				Properties: map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+					propertyprovider.NodeCountProperty: {
+						Value: "1",
+					},
+				},
+				ResourceUsage: clusterv1beta1.ResourceUsage{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("10"),
+						corev1.ResourceMemory: resource.MustParse("10Gi"),
+					},
+					Allocatable: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("8"),
+						corev1.ResourceMemory: resource.MustParse("8Gi"),
+					},
+					Available: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("5"),
+						corev1.ResourceMemory: resource.MustParse("5Gi"),
+					},
+				},
+			},
+		},
+		{
+			name:             "provider starts up successfully",
+			propertyProvider: &dummyProvider{},
+			wantIMCStatus: clusterv1beta1.InternalMemberClusterStatus{
+				Properties: map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+					exampleClusterPropertyName: {
+						Value: exampleClusterPropertyValue,
+					},
+				},
+				ResourceUsage: clusterv1beta1.ResourceUsage{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("10"),
+						corev1.ResourceMemory: resource.MustParse("10Gi"),
+					},
+					Allocatable: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("5"),
+						corev1.ResourceMemory: resource.MustParse("5Gi"),
```

Comment:
IIRC, this is tied to dummyprovider, it would be good to make it more explicit, or the test is easy to be broken.

Reply:
Done.